lib/http-context.js

```diff
@@ -192,27 +191,45 @@ function coerce(str) {
 // coerce every string in the given object / array
 function coerceAll(obj) {
   var type = Array.isArray(obj) ? 'array' : typeof obj;
+  var i;
+  var n;
 
   switch (type) {
     case 'string':
       return coerce(obj);
     case 'object':
       if (obj) {
-        Object.keys(obj).forEach(function(key) {
+        var props = Object.keys(obj);
```

Comment:
Petka is not wrong but it's a tradeoff.  Garbage collection is frequently the biggest performance killer, hence my advice.  Try tracing it with --trace_opt and --trace_deopt.  If it does deoptimize, move it into a separate function.